atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/TargetedSweepMetrics.java

```diff
@@ -116,45 +117,54 @@ private static Long minimum(Collection<Long> currentValues) {
-        private final AccumulatingValueMetric tombstonesPut = new AccumulatingValueMetric();
-        private final AccumulatingValueMetric abortedWritesDeleted = new AccumulatingValueMetric();
-        private final CurrentValueMetric<Long> sweepTimestamp = new CurrentValueMetric<>();
-        private final AggregatingVersionedSupplier<Long> lastSweptTsSupplier;
-        private final AtomicLong lastWrittenTs = new AtomicLong(0L);
+        private final MetricsManager manager;
+        private final TimestampedAccumulatingValueMetric enqueuedWrites;
+        private final AccumulatingValueMetric entriesRead;
+        private final AccumulatingValueMetric tombstonesPut;
+        private final AccumulatingValueMetric abortedWritesDeleted;
+        private final CurrentValueMetric<Long> sweepTimestamp;
+        private final AggregatingVersionedMetric<Long> lastSweptTs;
 
         private MetricsForStrategy(MetricsManager manager, String strategy, Function<Long, Long> tsToMillis,
                 Clock wallClock, long recomputeMillis) {
             Map<String, String> tag = ImmutableMap.of(AtlasDbMetricNames.TAG_STRATEGY, strategy);
-            register(manager, AtlasDbMetricNames.ENQUEUED_WRITES, enqueuedWrites, tag);
-            register(manager, AtlasDbMetricNames.ENTRIES_READ, entriesRead, tag);
-            register(manager, AtlasDbMetricNames.TOMBSTONES_PUT, tombstonesPut, tag);
-            register(manager, AtlasDbMetricNames.ABORTED_WRITES_DELETED, abortedWritesDeleted, tag);
-            register(manager, AtlasDbMetricNames.SWEEP_TS, sweepTimestamp, tag);
+            this.manager = manager;
+            enqueuedWrites = register(AtlasDbMetricNames.ENQUEUED_WRITES, TimestampedAccumulatingValueMetric.create(),
+                    tag);
+            entriesRead = registerAccumulating(AtlasDbMetricNames.ENTRIES_READ, tag);
+            tombstonesPut = registerAccumulating(AtlasDbMetricNames.TOMBSTONES_PUT, tag);
+            abortedWritesDeleted = registerAccumulating(AtlasDbMetricNames.ABORTED_WRITES_DELETED, tag);
+            sweepTimestamp = register(AtlasDbMetricNames.SWEEP_TS, new CurrentValueMetric<>(), tag);
+
+            AggregatingVersionedSupplier<Long> lastSweptTsSupplier = new AggregatingVersionedSupplier<>(
+                    TargetedSweepMetrics::minimum, recomputeMillis);
+            lastSweptTs = register(AtlasDbMetricNames.LAST_SWEPT_TS,
+                    new AggregatingVersionedMetric<>(lastSweptTsSupplier), tag);
 
-            lastSweptTsSupplier = new AggregatingVersionedSupplier<>(TargetedSweepMetrics::minimum, recomputeMillis);
             Supplier<Long> millisSinceLastSweptTs = new CachedComposedSupplier<>(
-                    lastSweptTs -> estimateMillisSinceTs(lastSweptTs, wallClock, tsToMillis),
-                    lastSweptTsSupplier);
+                    swept -> estimateMillisSinceTs(swept, wallClock, tsToMillis), lastSweptTs::getVersionedValue);
```

Comment:
nit: I know there are many `lastSwept*` variables but if possible could this be named different than `swept`, maybe `sweptTs`